Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,6 +140,164 @@ public class PushDeploymentController : ApiController
             }
         }
 
+        [HttpPost]
+        [HttpPut]
+        public async Task<HttpResponseMessage> OnePushDeploy(
+            [FromUri] bool isAsync = false,
+            [FromUri] string author = null,
+            [FromUri] string authorEmail = null,
+            [FromUri] string deployer = OneDeploy,
+            [FromUri] string message = DefaultMessage)
+        {
+            using (_tracer.Step("OnePushDeploy"))
+            {
+                var response = Request.CreateResponse();
+                //Checks the "type" query parameter, defaults to "static" if no query parameter is provided 
+               var deployType = Request.RequestUri.ParseQueryString()["type"];
+                if (string.IsNullOrWhiteSpace(deployType))
+                {
+                    deployType = "static";
+                }
+                var website_framework = (System.Environment.GetEnvironmentVariable("website_framework".ToLower())).ToLower();
+                //Checks the "path" queary parameter, path root at /wwwroot TBD if need to change to /home
+                string path = Request.RequestUri.ParseQueryString()["path"];
+                //Checks the "clean" query parameter, defauts to False if no query parameter is provided
+                var cleanQuery = Request.RequestUri.ParseQueryString()["clean"];
+                bool isCleanDeployment = Convert.ToBoolean(cleanQuery);
+                //Checks the "async" query parameter, defaults to false if no query parameter is provided
+                var asyncQuery = Request.RequestUri.ParseQueryString()["async"];
+                isAsync = Convert.ToBoolean(asyncQuery);
+
+                var deploymentInfo = new ZipDeploymentInfo(_environment, _traceFactory)
+                {
+                    AllowDeploymentWhileScmDisabled = true,
+                    Deployer = deployer,
+                    WatchedFilePath = Path.Combine("WEB-INF", "web.xml"),
```

Comment:
>WatchedFilePath = Path.Combine("WEB-INF", "web.xml"), [](start = 20, length = 53)

This is meant as a way to trigger a restart by touching the watched file. Not needed as you have your own way of triggering a restart.